src/interfaces.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{ActivityCreatorId, ActivityId};
+use near_contract_standards::non_fungible_token::metadata::{NFTContractMetadata, TokenMetadata};
+use near_sdk::AccountId;
+
+pub trait OwnerAction {
+    fn set_contract_metadata(&mut self, metadata: NFTContractMetadata) -> bool;
+
+    fn add_creator_whitelist(&mut self, account_ids: Vec<AccountId>);
+
+    fn remove_creator_whitelist(&mut self, account_ids: Vec<AccountId>);
+}
+
+pub trait ActivityCreatorAction {
```

Comment:
I can't find a function to create activity. Maybe it's a mistake?

Reply:
This method: create_activity_token_metadata